sigstore/_cli.py

```diff
@@ -228,6 +238,102 @@ def _parser() -> argparse.ArgumentParser:
         help="the operation to perform",
     )
 
+    # `sigstore attest`
+    attest = subcommands.add_parser(
+        "attest",
+        help="sign one or more inputs using DSSE",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        parents=[parent_parser],
+    )
+    attest.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    dsse_options = attest.add_argument_group("DSSE options")
+    dsse_options.add_argument(
+        "--predicate",
+        metavar="FILE",
+        type=Path,
+        required=True,
+        help="Path to the predicate file",
+    )
+    dsse_options.add_argument(
+        "--predicate-type",
+        metavar="TYPE",
+        choices=PREDICATE_TYPES_CLI_MAP,
```

Comment:
Out of curiosity: do other clients use these shorthand names? I'm wondering if we should just use the predicate URLs directly as choices here -- that makes the CLI slightly less ergonomic, but avoids adding one more custom convention 🙂

Reply:
Yes, I got these from `cosign`:
```
    --type='custom':
        specify a predicate type
        (slsaprovenance|slsaprovenance02|slsaprovenance1|link|spdx|spdxjson|cyclonedx|vuln|openvex|custom)
        or an URI
```

Reply:
Ah my bad, I forgot I had changed them. I changed them to `slsaprovenance0_2` and `slsaprovenance1_0` because `slsaprovenance1` seems ambiguous. But yeah, these are not standard in any way

Reply:
Great! Makes perfect sense to use them, then.

CC @loosebazooka @haydentherapper for visibility -- this is maybe too much of an implementation detail to standardize, but something for other clients to be aware of 🙂

Reply:
Seems reasonable to me to specify the predicate type as its full URL and not deal with shorthand.